jenkinsci/jenkins-multijob-plugin
# [JENKINS-73965] Extract event handlers in `MultiJobTestResults/summary.jelly`

[JENKINS-73965](https://issues.jenkins.io/browse/JENKINS-73965)

Removes inline JS and onClick. 

### Testing done

1. Create two freestyle projects. In the build section, add to each a shell script to create jUnit test results (need at least 4-5 failures total to show the relevant expand link. 
ex:
```
# Create the directory for storing test results
mkdir -p test-results

# Generate a JUnit XML report with 6 tests: 3 passed, 3 failed
cat <<EOF > test-results/test-report-project2.xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="TestSuiteProject2" tests="6" failures="3" time="0.005">
  <testcase classname="TestClass1" name="testSuccess1" time="0.001"/>
  <testcase classname="TestClass2" name="testSuccess2" time="0.001"/>
  <testcase classname="TestClass3" name="testSuccess3" time="0.001"/>

  <testcase classname="TestClass4" name="testFailure1" time="0.001">
    <failure message="Test failure 1">Assertion failed in testFailure1</failure>
  </testcase>
  
  <testcase classname="TestClass5" name="testFailure2" time="0.001">
    <failure message="Test failure 2">Null pointer exception in testFailure2</failure>
  </testcase>
  
  <testcase classname="TestClass6" name="testFailure3" time="0.001">
    <failure message="Test failure 3">Array index out of bounds in testFailure3</failure>
  </testcase>
</testsuite>
EOF
```
2. in the post-build section for each freestyle project, select `Publish JUnit test result report` and enter the file src from example: `test-results/*.xml`
3. Create new MultiJob Project.
4. in the build steps select `MultiJob Phase`
5. Add the two Phase Jobs (the two freestyle jobs just created). In the advanced section select `Aggregate Test Results`
6. In the Post Build Actions, select `Aggregate downstream test results` and check the both boxes available. 
7. Build the project and navigate to `Status` section of the latest build. This brings you to the relevant screen. 

[Status Quo Video](https://www.loom.com/share/81f9a3a06dcb4e5089462fa62c3264a3)
[CSP Report Only](https://www.loom.com/share/2ebcfa68471d40a69d3e70e648cf7902)
[CSP Restrictive](https://www.loom.com/share/b5879a75e0b64b638134e150cc0b2c0d)



### Submitter checklist
